roles/kubernetes/master/templates/manifests/kube-apiserver.manifest.j2

```diff
@@ -100,6 +100,16 @@ spec:
 {% if kube_feature_gates %}
     - --feature-gates={{ kube_feature_gates|join(',') }}
 {% endif %}
+{% if kube_version | version_compare('1.9', '>=') %}
+    - --requestheader-client-ca-file=/etc/kubernetes/ssl/ca.pem
+    - --requestheader-allowed-names=system:aggregator-proxy-client
+    - "--requestheader-extra-headers-prefix=X-Remote-Extra-"
+    - --requestheader-group-headers=X-Remote-Group
+    - --requestheader-username-headers=X-Remote-User
+    - --enable-aggregator-routing=true
```

Comment:
Let's make this configurable. Maybe some people don't want it enabled.
Add to roles/kubernetes/master/defaults/main.yml:
`kube_api_aggregator_routing: true`
Change this line to:
`    - --enable-aggregator-routing={{ kube_api_aggregator_routing }}`